lib/algolia_index.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'algolia_index/models/firm'
+require_relative 'algolia_index/models/adviser'
+require_relative 'algolia_index/models/office'
+
+require_relative 'algolia_index/serializers/firm_serializer'
+require_relative 'algolia_index/serializers/adviser_serializer'
+require_relative 'algolia_index/serializers/office_serializer'
+
+module AlgoliaIndex
+  INDICES = {
+    advisers: "firm-advisers#{'-test' if Rails.env.test?}",
+    offices: "firm-offices#{'-test' if Rails.env.test?}"
+  }.freeze
+
+  class << self
+    def handle_update!(klass:, id:, firm_id:)
+      record = "AlgoliaIndex::#{klass}".constantize.new(
+        klass: klass,
+        id: id,
+        firm_id: firm_id
+      )
+      if record.exists?
+        record.update!
+      else
+        record.destroy!
+      end
+    end
+
+    def index_advisers
```

Comment:
Perhaps `indexed_advisers` instead - reads more like a noun that way, current form reads like a verb.

Same for offices below.

Reply:
@malcolmbaig that's true! I think at some point we were using  `advisers_index` and `offices_index` but somehow we ended reversing the order.